public_html/wp-content/plugins/pattern-directory/includes/favorite.php

```diff
@@ -84,3 +84,81 @@ function get_favorites( $user = 0 ) {
 
 	return array_map( 'absint', $favorites );
 }
+
+/**
+ * Get the cached count of how many times this pattern has been favorited.
+ *
+ * @param int|WP_Post $post The pattern to check.
+ * @return integer
+ */
+function get_favorite_count( $post = 0 ) {
+	$post = get_block_pattern( $post );
+	if ( ! $post ) {
+		return false;
+	}
+
+	return absint( get_post_meta( $post->ID, META_KEY, true ) );
+}
+
+/**
+ * Get a count of how many times this pattern has been favorited, directly from the users table.
+ *
+ * @param int|WP_Post $post The pattern to check.
+ * @return integer
+ */
+function get_raw_favorite_count( $post = 0 ) {
+	global $wpdb;
+	$post = get_block_pattern( $post );
+	if ( ! $post ) {
+		return false;
+	}
+	$count = $wpdb->get_var( $wpdb->prepare(
+		"SELECT COUNT(*)
+			FROM {$wpdb->usermeta}
+			WHERE meta_key=%s
+			AND meta_value=%d",
+		META_KEY,
+		$post->ID
+	) );
+
+	return absint( $count );
+}
+
+/**
+ * Update a given post's favorite count cache.
+ *
+ * @param mixed $value The post ID.
+ */
+function update_favorite_cache( $value ) {
+	if ( ! is_numeric( $value ) ) {
+		return;
+	}
+
+	$count = get_raw_favorite_count( $value );
+	update_post_meta( $value, META_KEY, $count );
+}
+
+/**
+ * Trigger the update of favorite count when a user favorites or unfavorites a pattern.
+ *
+ * @param int    $mid         The meta ID.
+ * @param int    $object_id   ID of the object metadata is for.
+ * @param string $meta_key    Metadata key.
+ * @param mixed  $_meta_value Metadata value. Serialized if non-scalar.
+ */
+function trigger_favorite_cache_update( $mid, $object_id, $meta_key, $_meta_value ) {
```

Comment:
Took me a minute to remember that `$object_id` is actually the user ID here, and `$_meta_value` is the post ID or IDs. Might be helpful to rename these to be more explicit.

Reply:
I'd prefer to keep these matching the core behavior - `trigger_favorite_cache_update` is called on every meta value, so `$_meta_value` isn't always the post ID. I updated `$object_id`, but want to leave `$_meta_value`.